test/reagenttest/testreagent.cljs

```diff
@@ -61,71 +66,74 @@
-
-        (r/set-state @self {:foo "you"})
-        (r/flush)
-        (is (= "hi you" (.-innerText div)))))
-    (is (= 4 @ran))))
+    (u/async
+      (p/do
+        (u/with-render
+          [comp]
+          (fn [div]
+            (p/do
+              (swap! ran inc)
+              (is (= "hi initial" (.-innerText div)))
+
+              (u/act (r/replace-state @self {:foo "there"}))
+              ;; (r/state @self)
+
+              ;; (r/flush)
+              (is (= "hi there" (.-innerText div)))
+
+              (u/act (r/set-state @self {:foo "you"}))
+              ;; (r/flush)
+              (is (= "hi you" (.-innerText div))))))
+        (is (= 4 @ran))))))
 
 (u/deftest ^:dom test-ratom-change
   (let [compiler u/*test-compiler*
         ran (r/atom 0)
         runs (rv/running)
         val (r/atom 0)
         secval (r/atom 0)
-        v1-ran (atom 0)
-        v1 (reaction (swap! v1-ran inc) @val)
+        reaction-ran (atom 0)
+        v1 (reaction (swap! reaction-ran inc) @val)
         comp (fn []
                (swap! ran inc)
                [:div (str "val " @v1 " " @val " " @secval)])]
-    (t/async done
-      (u/with-mounted-component-async [comp]
-        (fn []
-          (r/next-tick
-            (fn []
-              (r/next-tick
-                (fn []
-                  (is (= runs (rv/running)))
-                  (is (= 2 @ran))
-                  (done))))))
-        compiler
-        (fn [C div done]
-          (r/flush)
-          (is (not= runs (rv/running)))
-          (is (= "val 0 0 0" (.-innerText div)))
-          (is (= 1 @ran))
-
-          (reset! secval 1)
-          (reset! secval 0)
-          (reset! val 1)
-          (reset! val 2)
-          (reset! val 1)
-          (is (= 1 @ran))
-          (is (= 1 @v1-ran))
-          (r/flush)
-          (is (= "val 1 1 0" (.-innerText div)))
-          (is (= 2 @ran) "ran once more")
-          (is (= 2 @v1-ran))
-
-          ;; should not be rendered
-          (reset! val 1)
-          (is (= 2 @v1-ran))
-          (r/flush)
-          (is (= 2 @v1-ran))
-          (is (= "val 1 1 0" (.-innerText div)))
-          (is (= 2 @ran) "did not run")
-          (done))))))
-
+    (u/async
+      (p/do
+        (u/with-render
+          [comp]
+          compiler
+          (fn [div]
+            ;; (r/flush)
+            (is (not= runs (rv/running)))
+            (is (= "val 0 0 0" (.-innerText div)))
+            (is (= 1 @ran))
+
+            (u/act
+              (reset! secval 1)
+              (reset! secval 0)
+              (reset! val 1)
+              (reset! val 2)
+              (reset! val 1))
+
+            ;; (r/flush)
+            (is (= "val 1 1 0" (.-innerText div)))
+            (is (= 2 @ran) "ran once more")
+            ;; NOTE: OKAY Here is a problem:
+            ;; reactions are now being run for each input ratom change because we don't have the queue anymore!
+            (is (= 2 @reaction-ran))
```

Comment:
As I was thinking, while components aren't rendered for each Ratom change due to the React queue, here the    `reaction` body is running each time input Ratoms change, which is potentially a (big) problem for re-frame.

Consider cases where text input value is used as a reaction (or subscription) input.